py-polars/src/lazyframe/mod.rs

```diff
@@ -65,17 +65,37 @@ impl PyLazyFrame {
         }
     }
 
-    #[cfg(all(feature = "json", feature = "serde_json"))]
-    fn serialize(&self, py_f: PyObject) -> PyResult<()> {
+    /// Serialize into binary data.
+    fn serialize_binary(&self, py_f: PyObject) -> PyResult<()> {
+        let file = BufWriter::new(get_file_like(py_f, true)?);
+        ciborium::into_writer(&self.ldf.logical_plan, file)
+            .map_err(|err| PyValueError::new_err(format!("{err:?}")))?;
+        Ok(())
+    }
+
+    /// Serialize into a JSON string.
+    #[cfg(feature = "json")]
+    fn serialize_json(&self, py_f: PyObject) -> PyResult<()> {
         let file = BufWriter::new(get_file_like(py_f, true)?);
         serde_json::to_writer(file, &self.ldf.logical_plan)
             .map_err(|err| PyValueError::new_err(format!("{err:?}")))?;
         Ok(())
     }
 
+    /// Deserialize a file-like object containing binary data into a LazyFrame.
+    #[staticmethod]
+    fn deserialize_binary(py_f: PyObject) -> PyResult<Self> {
+        let file = get_file_like(py_f, false)?;
+        let reader = BufReader::new(file);
+        let lp = ciborium::from_reader::<DslPlan, _>(reader)
+            .map_err(|err| PyValueError::new_err(format!("{err:?}")))?;
+        Ok(LazyFrame::from(lp).into())
+    }
```

Comment:
This implementation fails for Enum types, sample error:
` ValueError: Semantic(None, "invalid type: string \"b\", expected a borrowed string")`

@ritchie46 Any idea what that might be? :thinking:

Reply:
Seems like a serde implementation error. Can you add a skipped test? Maybe we can fix it before 1.0, but I rather have the API correct first.

In any case, I will take a look if you have a repro.

Reply:
Chiel solved it in one line :+1: